src/libs/TransactionUtils.ts

```diff
@@ -307,11 +308,30 @@ function getOriginalAmount(transaction: Transaction): number {
     return Math.abs(amount);
 }
 
+/**
+ * Verify if the transaction is of Distance request and is not fully ready:
+ * - it has a zero amount, which means the request was created offline and expects the distance calculation from the server
+ * - it is in `isLoading` state, which means the waypoints were updated offline and the distance requires re-calculation
+ */
+function isLoadingDistanceRequest(transaction: OnyxEntry<Transaction>): boolean {
+    if (!transaction) {
+        return false;
+    }
+
+    const amount = getAmount(transaction, false);
+    return isDistanceRequest(transaction) && (!!transaction?.isLoading || amount === 0);
+}
+
 /**
  * Return the merchant field from the transaction, return the modifiedMerchant if present.
  */
 function getMerchant(transaction: OnyxEntry<Transaction>): string {
-    return transaction?.modifiedMerchant ? transaction.modifiedMerchant : transaction?.merchant ?? '';
+    if (!transaction) {
+        return '';
+    }
+
+    const merchant = transaction.modifiedMerchant ? transaction.modifiedMerchant : transaction.merchant ?? '';
+    return isLoadingDistanceRequest(transaction) ? merchant.replace(CONST.REGEX.FIRST_SPACE, Localize.translateLocal('common.tbd')) : merchant;
```

Comment:
Yes, I get your point and tend to agree with it. But it will mean that we need to have duplicate code scattered across components. Let me bring this to Slack, I think it's worth a broader discussion

Reply:
My thought process here is that we are calculating the `merchant` based only on the data present on the `transaction` object in Onyx. We are not using any external (state) params so we cannot call this data state-dependant (or reactive) in its full meaning